src/panels/lovelace/editor/config-elements/hui-button-card-editor.ts

```diff
@@ -6,24 +6,39 @@ import {
   internalProperty,
   LitElement,
   property,
+  query,
   TemplateResult,
 } from "lit-element";
 import { assert, boolean, object, optional, string } from "superstruct";
-import { fireEvent } from "../../../../common/dom/fire_event";
+
+import { fireEvent, HASSDomEvent } from "../../../../common/dom/fire_event";
 import { stateIcon } from "../../../../common/entity/state_icon";
 import { computeRTLDirection } from "../../../../common/util/compute_rtl";
-import "../../../../components/ha-formfield";
-import "../../../../components/ha-icon-input";
-import "../../../../components/ha-switch";
 import { ActionConfig } from "../../../../data/lovelace";
 import { HomeAssistant } from "../../../../types";
 import { ButtonCardConfig } from "../../cards/types";
-import "../../components/hui-action-editor";
-import "../../components/hui-entity-editor";
-import "../../components/hui-theme-select-editor";
 import { LovelaceCardEditor } from "../../types";
-import { actionConfigStruct, EditorTarget } from "../types";
+import {
+  actionConfigStruct,
+  EditorTarget,
+  GUIModeChangedEvent,
+} from "../types";
 import { configElementStyle } from "./config-elements-style";
+// eslint-disable-next-line import/no-duplicates
+import { HuiElementEditor } from "../hui-element-editor";
+// eslint-disable-next-line import/no-duplicates
+import { EditActionEvent } from "../../components/hui-actions-editor";
```

Comment:
```suggestion
import type { HuiElementEditor } from "../hui-element-editor";
// eslint-disable-next-line import/no-duplicates
import type { EditActionEvent } from "../../components/hui-actions-editor";
```

Reply:
I get TS errors if I remove that

Reply:
You are missing my change. Add the type to those